src/simulation/LoadGenerator.cpp

```diff
@@ -244,20 +372,56 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
             MIN_UNIQUE_ACCOUNT_MULTIPLIER);
     }
 
-    if (cfg.mode == LoadGenMode::SOROBAN &&
+    if ((cfg.mode == LoadGenMode::SOROBAN_INVOKE ||
+         cfg.mode == LoadGenMode::SOROBAN_WASM) &&
         protocolVersionIsBefore(mApp.getLedgerManager()
                                     .getLastClosedLedgerHeader()
                                     .header.ledgerVersion,
                                 SOROBAN_PROTOCOL_VERSION))
     {
-        errorMsg = "Soroban mode requires protocol version 20 or higher";
+        errorMsg = "Soroban modes require protocol version 20 or higher";
+    }
+
+    if (cfg.mode == LoadGenMode::SOROBAN_INVOKE)
+    {
+        LedgerTxn ltx(mApp.getLedgerTxnRoot());
+        auto& sorobanCfg = mApp.getLedgerManager().getSorobanNetworkConfig(ltx);
+        if (cfg.nDataEntries == 0)
```

Comment:
These checks don't make much sense even with the current way to setup loadgen - why shouldn't I be able to skip the storage phase completely (and with suggested refactoring it would make even less sense).